plat_all-unix-style.go

```diff
@@ -121,9 +121,18 @@ func (task *TaskRun) EnvVars() []string {
 		taskEnv[k] = v
 	}
 	taskEnv["TASK_ID"] = task.TaskID
+	taskEnv["TASKCLUSTER_ROOT_URL"] = config.RootURL
+
+	// see if we should include TASKCLUSTER_PROXY_URL
+	tcProxyFeature := &TaskclusterProxyFeature{}
+	if tcProxyFeature.IsEnabled(task) {
+		taskEnv["TASKCLUSTER_PROXY_URL"] = tcProxyFeature.getTaskclusterProxyURL()
+	}
```

Comment:
nit: This should really be done inside the feature (i.e. inside `taskcluster_proxy.go`). Extra points for implementing a generic method to augment an environment, which uses `win32.MergeEnvLists` on Windows, or just appends to the env for non-windows (like you had before). This could be in `plat_windows.go` and `plat_all-unix-style.go`, for example.

Note, with the suggested fix from my previous comment, `env` is always expanded to the full environment (i.e. not nil) so augmenting it suffices (no need to consider case of `env == nil`).

If you don't feel inclined to do this, I'm happy to implement it in a follow-up PR which I can send to you for review. Your version works correctly, it just has the architectural disadvantage that the core now has taskcluster proxy feature concerns.

Reply:
I did this.  I like the design a good bit more.  I did it in Start instead of NewTaskFeature because that method can return an error.